docs/structured-streaming-programming-guide.md

```diff
@@ -1792,7 +1792,85 @@ hence the number is not same as the number of original input rows. You'd like to
 There's a known workaround: split your streaming query into multiple queries per stateful operator, and ensure
 end-to-end exactly once per query. Ensuring end-to-end exactly once for the last query is optional.
 
-### State Store and task locality
+### State Store
+
+State store is a versioned key-value store which provides both read and write operations. In
+structured streaming, we use the state store provider to handle the stateful operations across
```

Comment:
You use `Structured Streaming` below. Either `Structured Streaming` or `structured streaming` are okay, just better to keep consistent.

Reply:
Make sense, done in 2de0e24